### PR TITLE
Never use a lock file for the build

### PR DESCRIPTION
**Pull request by blink1073**
Because otherwise we can end up with unwanted duplicates
cf https://github.com/yarnpkg/yarn/issues/3967

### CODE REVIEW

**Comment by blink1073** [comment]
cc @wolfv

**Comment by wolfv** [comment]
thanks for the fix. 
I should then probably also remove the yarn.lock file from my repository?

**Comment by blink1073** [comment]
I think the lock file is still useful for a library author, it just doesn't work well for us yet for creating a fully deduped application.

File: docs/source/user/extensions.rst [hunk 1]
```diff
@@ -133,6 +133,11 @@ Building consists of:
 -  Bundling the assets
 -  Copying the bundled assets to the ``static`` directory
 
+Note that building will always use the latest JavaScript packages that meet
+the dependency requirements of JupyterLab itself any installed extensions.
```
Comment (jasongrout) [inline docs/source/user/extensions.rst]: itself *and* any installed ....

File: docs/source/developer/extension_dev.rst [hunk 2]
```diff
@@ -145,9 +145,12 @@ the CSS files) are watched by the WebPack process.
 Note that the application is built against **released** versions of the
 core JupyterLab extensions. If your extension depends on JupyterLab
 packages, it should be compatible with the dependencies in the
-``jupyterlab/static/package.json`` file. If you must install a extension
-into a development branch of JupyterLab, you have to graft it into the
-source tree of JupyterLab itself. This may be done using the command
+``jupyterlab/static/package.json`` file.  Note that building will always use the latest JavaScript packages that meet the dependency requirements of JupyterLab itself any installed extensions.  If you wish to test against a
```
Comment (jasongrout) [inline docs/source/developer/extension_dev.rst]: itself *and* any installed ....

**Comment by jasongrout** [comment]
Looks good to me, modulo adding two missing words in the docs.

**Comment by blink1073** [comment]
Ha, sick brain strikes again.

**Comment by jasongrout** [comment]
I'll test and then merge if it works okay too.

**Comment by jasongrout** [comment]
Works well for me too. Thanks!

**Comment by jasongrout** [comment]
See also discussion around this: https://gitter.im/jupyterlab/jupyterlab?at=5a7b04dee217167e2c53f1c3 (and before)

> okay, here’s what is going on. In the yarn.lock file we ship with 0.31.3, it has an entry for @jupyterlab/application@^0.15.3. We use that as a basis when adding extensions. In this case running yarn again picks up an entry for @jupyterlab/application@^0.15.0 that resolves to the latest published 0.15.4, but does not dedupe the existing entry for 0.15.3 in the lock file. When I switch to jlab 0.31.4, we get the single entry because it already had 0.15.4 resolved in the lock file

**Pull request merged**